test/test_timer.rs

```diff
@@ -82,9 +82,12 @@ fn alarm_fires() {
         }
     }
 
-    // Replace the old signal handler now that we've completed the test. If the
-    // test fails this process panics, so the fact we might not get here is
-    // okay.
+    // Cleanup:
+    // 1) deregister the OS's timer.
+    // 2) Replace the old signal handler now that we've completed the test. If
+    //    the test fails this process panics, so the fact we might not get here
+    //    is okay.
+    drop(timer);
```

Comment:
This doesn't seem to address the issue on my NetBSD VM. I'm not entirely sure why, but I think it might be related to the following section of the timer_delete man page:

> Pending notification events (signals) may or may not be delivered.

Reply:
Adding a 10ms delay after dropping but before removing the signal handler seems to provide enough time for things to resolve on my VM.

Reply:
Interesting.  From the FreeBSD man page: "Pending signals for the deleted timer are cleared." and from OpenGroup: "The disposition of pending signals for the deleted timer is unspecified."  So I guess OpenGroup is the lowest common denominator.  How do we ensure that all pending signals are dealt with?

Reply:
I tried `thread::yield_now()` without success, so I think you're current timing solution is as good as we'll get for now.

I'm going to leave the tests running for a bit, and if they don't fail with the current timeout for a few minutes I'll merge this.

Reply:
Oh, hmm. I suppose this implementation could destroy the timer, replace the old signal handler and then call `sigpending`. If the result of `sigpending` turns up empty the previously deleted timer didn't leave any trailing signals behind on systems that don't clean up. I'm not totally convinced that the OpenGroup wording doesn't allow for some unexpected ordering of operations in that proposed implementation, that said. The implementation that was merged up here makes good sense to me. 

Apologies for not noticing this myself.

Reply:
I don't think `sigpending` will work, because of this line from the man page:
```
Signals may be pending because they are currently
masked, or transiently before delivery (although the latter case is not
normally detectable).
```

Reply:
Ah ha, I believe you are correct.